src/TheNextWeek/main.cc

```diff
@@ -20,272 +20,307 @@
 #include "material.h"
 #include "moving_sphere.h"
 #include "sphere.h"
-
 #include <iostream>
 
 
-vec3 ray_color(const ray& r, hittable *world, int depth) {
+vec3 ray_color(const ray& r, hittable& world, int depth) {
     hit_record rec;
-    if (depth <= 0 || !world->hit(r, 0.001, infinity, rec))
+
+    // If we've exceeded the ray bounce limit, no more light is gathered.
+    if (depth <= 0)
         return vec3(0,0,0);
 
+    // If the ray hits nothing, return the background color.
+    if (!world.hit(r, 0.001, infinity, rec))
+        return vec3(0.5,0.5,0.5);
```

Comment:
I understand why this change exists (having grey background).
I just don't love it, and having a scene-specific background was a big reason to create a dedicated `scene` class (for me).

At least as an intermediate step, create a `scene_background` function that is `inline`d and returns vec3(0.5)

Reply:
1. Make the change easy.
2. Make the easy change.

First step is to have code for a background color. Next step is to use it in the `ray_color` function. Final step is to get the background color from the scene/render-params. This is just the first iteration.